index.js

```diff
@@ -113,6 +115,21 @@ class PQueue {
 		return Promise.all(fns.map(fn => this.add(fn, opts)));
 	}
 
+	start() {
+		if (!this._paused) {
+			return;
+		}
+
+		this._paused = false;
+		while (this.queue.size > 0 && this._pendingCount < this._concurrency) {
```

Comment:
Do we need this `while` loop? I thought `run` method would automatically call `_next` which should do the same thing.

Reply:
`run` method will not be called if it's paused (it'll only be enqueued). 
So when you start (or restart), it should loop and dequeue.